### ISSUE TITLE
pgvector support

### ISSUE THREAD

**Issue by retendo** [Zaid-Ajaj/Npgsql.FSharp#139]
Is there anything planned already? Otherwise I could come up with something, as we need it anyway for our project.

I would probably have a look at https://github.com/pgvector/pgvector-dotnet or even add it as a dependency if that would be ok.

**Comment by Zaid-Ajaj** [comment]
There is nothing planned for this. It is not already possible to use this without taking a dependency on the library? Because looking at the docs:
```cs
var dataSourceBuilder = new NpgsqlDataSourceBuilder(connString);
dataSourceBuilder.UseVector();
await using var dataSource = dataSourceBuilder.Build();

var conn = dataSource.OpenConnection();
```
Since it gives you a `NpgsqlConnection` , you can use that from this library with 
```fs
conn
|> Sql.existingConnection
```

Specialized input parameters can be implemented with `Sql.parameter customParameter` and reading rows should be possible because the `Sql.execute` and friends give you access to the underlying `NgpsqlReader`

**Comment by retendo** [comment]
@Zaid-Ajaj Yes, it's definitely possible. I was just curious if it is something you are interested in having "first-class support" for. If not, then your answer will probably be sufficient for anyone looking to use pgvector with Npgsql.FSharp. 👍
